### DISCUSSION TITLE
Castle screen, Troop arrangement, make using the mouse wheel up and the mouse wheel down perform the same actions as the arrow up and arrow down keys on the keyboard

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10269] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

In the castle screen, make using the mouse wheel up and the mouse wheel down perform the same actions as the arrow up and arrow down keys on the keyboard.

In fheroes2:
Using keyboard keys:

https://github.com/user-attachments/assets/ae3399c3-8113-48da-b378-6cfe5f0a9320

### Describe the possible solution

Make using the mouse wheel up and the mouse wheel down perform the same actions as the arrow up and arrow down keys on the keyboard.

**Comment by zenseii** [comment]
Hi, @LeHerosInconnu. I believe this could be solved when this issue https://github.com/ihhub/fheroes2/issues/6580 gets solved because allowing to map hotkeys to the extra buttons on the mouse would allow you to map this specific function to the mouse wheel.

As a consequence, I believe this issue could be closed since it can be considered as part of what that other issue covers. What do you think, @LeHerosInconnu ?

**Comment by LeHerosInconnu** [comment]
Hello @zenseii,

> Hi, @LeHerosInconnu. I believe this could be solved when this issue #6580 gets solved because allowing to map hotkeys to the extra buttons on the mouse would allow you to map this specific function to the mouse wheel.
> 
> As a consequence, I believe this issue could be closed since it can be considered as part of what that other issue covers. What do you think, @LeHerosInconnu ?

I prefer to leave the issue open, as it allows me to see what has already been done or what is still to come.
It's not always easy to remember what's already been suggested over the years. :)
I have provided a link to this issue in the indicated thread.

**Comment by zenseii** [comment]
@LeHerosInconnu 

> I prefer to leave the issue open, as it allows me to see what has already been done or what is still to come. It's not always easy to remember what's already been suggested over the years. :) 

I basically just did a search for "scroll wheel" in the issues search box because I imagined such a suggestion would have already been made by now😅

>I have provided a link to this issue in the indicated thread.

As you can see from the issue thread, just from me mentioning it by a link here it already got linked automatically. This is just fyi.

**Comment by ihhub** [comment]
I am moving this to Discussion topic. It is not very intuitive change. Mouse wheel is usually used for scrolling. Here we are trying to use it for different purposes.

With recent changes we also allow to use additional mouse buttons to be mapped as hot keys.
